runtime/scripts/math.js

```diff
@@ -2051,6 +2051,124 @@ var math = Numbas.math = /** @lends Numbas.math */ {
             }
         }
         return factors;
+    },
+           /** Combines two matrices horizontally
+     * 
+     * @param {matrix} m1
+     * @param {matrix} m2
+     * @returns {matrix}
+     */
+            combine_horizontally: function(m1,m2){
+                m1.columns = m1[0].length;
```

Comment:
`m1` and `m2` should already have `rows` and `columns` properties: that's the point of having them!

It's not a good idea to repeat this logic for working out the dimensions of a matrix everywhere that they're used. If you say that the arguments have type `matrix`, you can assume that they have these properties.